app/src/main/java/com/example/mobiledger/data/sources/room/categories/CategoryDao.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.mobiledger.data.sources.room.categories
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.example.mobiledger.domain.entities.ExpenseCategoryListEntity
+import com.example.mobiledger.domain.entities.IncomeCategoryListEntity
+
+@Dao
+interface CategoryDao {
+
+    @Query("SELECT count(*) FROM category")
+    suspend fun hasCategory(): Int?
```

Comment:
I think instead of has category, there should be has income category and has expense category

Reply:
Discussed for now, will change in future if required.